openqabot/loader/gitea.py

```diff
@@ -418,12 +418,17 @@ def make_incident_from_pr(
         if len(incident["channels"]) == 0:
             log.info("Skipping PR %s, no channels found/considered", number)
             return None
-        if (
-            only_successful_builds
-            and len(incident["failed_or_unpublished_packages"]) > 0
-        ):
-            log.info("Skipping PR %s, not all packages succeeded and published", number)
-            return None
+        if only_successful_builds:
+            if len(incident["failed_or_unpublished_packages"]) > 0:
+                log.info(
+                    "Skipping PR %s, not all packages succeeded and published", number
+                )
+                return None
+            if len(incident["successful_packages"]) < 1:
+                log.info(
+                    "Skipping PR %s, no packages have been built/published", number
+                )
+                return None
```

Comment:
I think this can be reworked by extracting a function and fulfilling the style request about not too many return statements

```suggestion
def packages_ok(incident):
    if len(incident["failed_or_unpublished_packages"]) > 0:
        log.info("Skipping PR %s, not all packages succeeded and published", number)
        return False
    if len(incident["successful_packages"]) < 1:
        log.info("Skipping PR %s, no packages have been built/published", number)
        return False
    return True

…
        if only_successful_builds and not packages_ok:
            return None
```

Reply:
I find that splitting this function makes it less readable but I can do that if you prefer this.

Reply:
Well, I prefer to not disable style rules. Same as the discussion about line length both you and me prefer to stay consistent. I consider this approach the same :)

Reply:
Well, commonly "too long functions" are seen as complicated and so are functions with "too many return statements". I think your change is good. Of course overall this adds about 2-3 lines but they are mostly blank lines for separation and the new function's declaration+call. Still I think an approach like this can help readability. Admittedly often a completely different approach can help much more not even getting into the challenge about "many return statements" but as neither you or me had a good idea right now how to do it I guess the current style is good enough